packages/react-components/react-tabs/src/components/TabList/useTabList.ts

```diff
@@ -83,13 +83,16 @@ export const useTabList_unstable = (props: TabListProps, ref: React.Ref<HTMLElem
       root: 'div',
     },
     root: slot.always(
-      getNativeElementProps('div', {
-        ref: useMergedRefs(ref, innerRef),
+      getIntrinsicElementProps('div', {
+        // FIXME:
+        // `ref` is wrongly assigned to be `HTMLElement` instead of `HTMLDivElement`
+        // but since it would be a breaking change to fix it, we are casting ref to it's proper type
+        ref: useMergedRefs(ref, innerRef) as React.Ref<HTMLDivElement>,
         role: 'tablist',
         'aria-orientation': vertical ? 'vertical' : 'horizontal',
         ...focusAttributes,
         ...props,
-      }),
+      } as const),
```

Comment:
Same question here, why do you have to cast the object as `const`?

Reply:
Same answer...
Because `aria-orientation` type is `'vertical' | 'horizontal'` and without const assertion TS will assume we're delivering `string` instead of `'vertical' | 'horizontal'`, which is a mistake that now we can get thanks to `getIntrinsicElementProps` 😁